src/ingest_validation_tools/upload.py

```diff
@@ -266,54 +330,86 @@ def _check_single_assay(self):
                 f"There is more than one TSV for this type: {', '.join(repeated)}"
             )
 
-    def _get_local_tsv_errors(self):
-        for path, schema in self.effective_tsv_paths.items():
-            if "data_path" not in schema.rows[0] or "contributors_path" not in schema.rows[0]:
-                self.errors.upload_metadata[f"{path} (as {schema.table_schema})"].append(
-                    "File is missing data_path or contributors_path."
-                )
-            for ref in [OtherTypes.CONTRIBUTORS, OtherTypes.ANTIBODIES]:
-                errors = self._get_ref_errors(ref, schema, path)
-                if errors:
-                    self.errors.upload_metadata.update(errors)
+    def _check_shared_upload(self):
+        self.is_shared_upload = {"global", "non_global"} == {
+            x.name
+            for x in self.directory_path.glob("*global")
+            if x.is_dir() and x.name in ["global", "non_global"]
+        }
 
-    def _get_directory_errors(self):
-        if self.is_multi_assay and self.multi_parent:
-            for data_path in self.multi_assay_data_paths:
-                dir_errors = self._check_data_path(
-                    self.multi_parent, Path(self.multi_parent.path), data_path
-                )
-                for schema in self.effective_tsv_paths.values():
-                    if not schema.dataset_type == self.multi_parent.dataset_type:
-                        schema.dir_schema = self.multi_parent.dir_schema
-                if dir_errors:
-                    self.errors.directory.update(dir_errors)
-        else:
-            for path, schema in self.effective_tsv_paths.items():
-                dir_errors = self._get_ref_errors("data", schema, path)
-                if dir_errors:
-                    self.errors.directory.update(dir_errors)
+    @cached_property
+    def shared_upload_non_global_paths(self) -> dict[str, list[tuple[str, str]]]:
```

Comment:
get all individual paths included in non_global_files (with tsv/row references)

part of fix for NIHDEV-809 investigate reference checking in shared uploads